### PR TITLE
Add HistoryPlugin to the Editor UI component

### LINKED ISSUE #8602: Make undo (ctrl+z) work in the Editor component
### Is your proposal related to a problem?

It would be great if the user can use the undo option while typing something. This option is missing in the Editor (@calcom\ui\components\editor) component.
The editor component is used for writing the description or agenda for a new meeting and in the About section of the user's profile
- https://app.cal.com/event-types?dialog=new&event
- https://app.cal.com/settings/my-account/profile

Current behavior:

https://user-images.githubusercontent.com/47350983/235458247-2250b14c-339a-4dab-aa06-f74c3870c1b4.mp4

### Describe the solution you'd like

The lexical react package comes with a [History](https://www.npmjs.com/package/@lexical/react/v/0.5.0#:~:text=%7D%20/%3E-,%3CHistoryPlugin%20/%3E,-%3CMyCustomAutoFocusPlugin%20/) Plugin which enables the undo/redo option.

Expected behavior (with the History plugin added):

https://user-images.githubusercontent.com/47350983/235457839-709f43d1-540e-4237-b74f-417593ccd93d.mp4

### PR DESCRIPTION
**Pull request by Praneeth2k**
## What does this PR do?

Fixes  https://github.com/calcom/cal.com/issues/8602
Adds the HistoryPlugin to the Editor component to enable undoing(ctrl+z) and redoing(ctrl+shift+z) of the changes in the editor

**Environment**: Staging(main branch) / Production

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- [x] Checked the Editor section, was able to undo/redo changes after adding the HistoryPlugin
- [x] yarn test-e2e

### CODE REVIEW

**Review by Udit-takkar — APPROVED** [review]
LGTM. Thanks for your contribution

**Comment by Praneeth2k** [comment]
@Udit-takkar Thanks for approving. I'm a bit fuzzy about what the next step is. I believe I need to wait for an approval from calcom/ui. I see that some tests have failed, should I be fixing them? It doesn't seem like those failures are related to my change.

**Comment by Udit-takkar** [comment]
@Praneeth2k i'll retrigger the tests

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**
